storage/innobase/buf/buf0buf.cc

```diff
@@ -2579,7 +2579,16 @@ buf_page_get_low(
 		in buf_page_t::read_complete() or
 		buf_pool_t::corrupted_evict(), or
 		after buf_zip_decompress() in this function. */
-		block->page.lock.s_lock();
+		if (rw_latch != RW_NO_LATCH) {
+			block->page.lock.s_lock();
+
+		/* For RW_NO_LATCH, we should not try to acquire S or X latch
+		directly as we could be violating the latching order resulting
+		in deadlock. Instead we try latching the page and retry in case
+		of a failure. */
+		} else if (!block->page.lock.s_lock_try()) {
+			goto wait_for_unzip;
+		}
```

Comment:
I would place the comment inside the `else if` body and indent it to the same level with the `goto` statement.

With this change, the `wait_for_unzip` label is no longer specific to the buffer pool management of `ROW_FORMAT=COMPRESSED` pages. Therefore, the label had better be renamed, or a new label be added next to it, such as `wait_for_read`.

Reply:
Done.